lib/src/parse/stylesheet.dart

```diff
@@ -815,19 +814,20 @@ abstract class StylesheetParser extends Parser {
   /// [start] should point before the `@`. [child] is called to consume any
   /// children that are specifically allowed in the caller's context.
   EachRule _eachRule(LineScannerState start, Statement child()) {
+    whitespace(allowNewlines: true);
     var wasInControlDirective = _inControlDirective;
     _inControlDirective = true;
 
     var variables = [variableName()];
-    whitespace();
+    whitespace(allowNewlines: false);
     while (scanner.scanChar($comma)) {
-      whitespace();
+      whitespace(allowNewlines: false);
       variables.add(variableName());
-      whitespace();
+      whitespace(allowNewlines: false);
```

Comment:
These should all allow newlines, because none of `@each $var`, `@each $var,` and `@each $var1, $var2` are valid in isolation.